interfaces/builtin/mpris.go

```diff
@@ -221,14 +224,40 @@ func (iface *MprisInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *i
 	}
 }
 
+func (iface *MprisInterface) GetName(attribs map[string]interface{}) (string, error) {
+	// default to snap name if 'name' attribute not set
+	mprisName := "@{SNAP_NAME}"
+	for attr := range attribs {
+		if attr != "name" {
+			return "", fmt.Errorf("unknown attribute '%s'", attr)
+		}
+		name, ok := attribs[attr].(string)
```

Comment:
Hmm, I don't see the default being set above. 

All I meant to say is that in `name, ok := attribs[attr].(string)` the `ok` will refer to the type assertion, not to the presence of the attribute. In the snippet I gave above you can differentiate between the missing attribute and the attribute being of the incorrect type.